contracts/Committee.sol

```diff
@@ -53,98 +86,132 @@ contract Committee is ICommittee, Ownable {
 
 	/// @dev Called by: Elections contract
 	/// Notifies a weight change for sorting to a relevant committee member.
-    /// weight = 0 indicates removal of the member from the committee (for exmaple on unregister, voteUnready, voteOut)
+    /// weight = 0 indicates removal of the member from the committee (for example on unregister, voteUnready, voteOut)
 	function memberWeightChange(address addr, uint256 weight) external onlyElectionsContract returns (bool committeeChanged, bool standbysChanged) {
-		if (isMember(addr)) {
-			members[addr].weight = weight;
-			return _rankValidator(addr);
+		MemberData memory memberData = membersData[addr];
+		if (memberData.isMember) {
+			memberData.weight = weight;
+			return _rankAndUpdateMember(Member({
+				addr: addr,
+				data: memberData
+			}));
 		}
 		return (false, false);
 	}
 
 	function memberReadyToSync(address addr) external onlyElectionsContract returns (bool committeeChanged, bool standbysChanged) {
-		if (isMember(addr)) {
-			members[addr].readyToSyncTimestamp = now;
-			members[addr].readyForCommittee = false;
-			return _rankValidator(addr);
+		MemberData memory memberData = membersData[addr];
+		if (memberData.isMember) {
+			memberData.readyToSyncTimestamp = now;
+			memberData.readyForCommittee = false;
+			return _rankAndUpdateMember(Member({
+				addr: addr,
+				data: memberData
+			}));
 		}
 		return (false, false);
 	}
 
 	function memberReadyForCommittee(address addr) external onlyElectionsContract returns (bool committeeChanged, bool standbysChanged) {
-		if (isMember(addr)) {
-			members[addr].readyToSyncTimestamp = now;
-			members[addr].readyForCommittee = true;
-			return _rankValidator(addr);
+		MemberData memory memberData = membersData[addr];
+		if (memberData.isMember) {
+			memberData.readyToSyncTimestamp = now;
```

Comment:
Heartbeat is not as specific, I see readyForCommittee as a more specific readyToSync, which makes this naming consistent.

Reply:
in this case I'd suggest throwing readyForCommittee away and replacing it with a boolean argument passed to  readyToSync. This would be consistent. We can do this change without affecting the elections contract API to maintain the external interface if you prefer. 

what do you think?

Reply:
Yes, I like it. But it requires changing the spec interface so let's ask @OdedWx what he thinks.

Reply:
Will add a TODO as to not block this PR